packages/pools/src/router/routes.ts

```diff
@@ -150,6 +155,55 @@ export class OptimizedRoutes implements TokenOutGivenInRouter {
 
     let routes = this.getCandidateRoutes(tokenIn.denom, tokenOutDenom);
 
+    // HOTFIX:
+    // Special case transmuter pool handling.
+    // Tranmuter pools provide a 1:1 swap with no slippage.
+    // As a result, if we see a transmuter in candidate routes,
+    // we can return it as a single pool swap.
+    for (let i = 0; i < routes.length; i++) {
+      const curRoute = routes[i];
+      if (curRoute.pools.length == 1) {
+        const singlePool = curRoute.pools[0];
+
+        const denomSize = singlePool.poolAssetDenoms.length;
+        if (denomSize != 2) {
+          continue;
+        }
+
+        const denomA = singlePool.poolAssetDenoms[0];
+        const denomB = singlePool.poolAssetDenoms[1];
+
+        // Confirm that token in is in the pool. If not skip.
+        if (tokenIn.denom != denomA && tokenIn.denom != denomB) {
+          continue;
+        }
+
+        // Confirm that token out is in the pool. If not skip.
+        if (tokenOutDenom != denomA && tokenOutDenom != denomB) {
+          continue;
+        }
+
+        // Confirm that this is a transmuter pool. If not skip.
+        const isTransmuterPoolSlice = transmuterPoolIDs.filter(
+          (poolId) => poolId == singlePool.id
+        );
+
+        if (isTransmuterPoolSlice.length == 0) {
+          continue;
+        }
+
+        // If all checks passed, use this pool as the only one in the route.
+
+        const directOutAmount = (
+          await this.calculateTokenOutByTokenIn([
+            { ...curRoute, initialAmount: tokenIn.amount },
+          ])
+        ).amount;
+
+        return [{ ...curRoute, initialAmount: directOutAmount }];
+      }
+    }
```

Comment:
Is there any reason why we don't check this directly on `this.getCandidateRoutes`?

Reply:
No reason. I will try now and see if it's a better locaton

Reply:
`getCandidateRoutes` is not  async.

I think it is better to keep this as is. The hotfix searches for transmuter pool in routes, if it finds one, it does an async calculation to determine the result and returns that as a substitute to the original split route right away.

Reply:
I like this. Thank you.

Applied:
[4345984](https://github.com/osmosis-labs/osmosis-frontend/pull/2184/commits/4345984e0f8dd8599db0c129c601875396d81cad)